app/services/course_search_service.rb

```diff
@@ -26,17 +41,9 @@ def call
       scope = scope.order(:distance)
     end
 
-    scope = scope.with_salary if funding_filter_salary?
-    scope = scope.with_qualifications(qualifications) if qualifications.any?
-    scope = scope.with_vacancies if has_vacancies?
-    scope = scope.with_study_modes(study_types) if study_types.any?
-    scope = scope.with_subjects(subject_codes) if subject_codes.any?
-    scope = scope.with_provider_name(provider_name) if provider_name.present?
-    scope = scope.with_send if send_courses_filter?
-    scope = scope.within(filter[:radius], origin: origin) if locations_filter?
-    scope = scope.with_funding_types(funding_types) if funding_types.any?
-
-    scope.distinct
+    scope
+    #  For some reason the api breaks if the 'scope.distinct' remains :-()
+    # scope.distinct
```

Comment:
??
```
Cause: PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: ..., $9, $10, $11, $12, $13, $14, $15, $16) ORDER BY CASE WHEN ...
```

Reply:
https://stackoverflow.com/questions/12693089/pgerror-select-distinct-order-by-expressions-must-appear-in-select-list

Reply:
its needed because of 

https://github.com/DFE-Digital/teacher-training-api/blob/826b00d4a6331cb6a36a7dfd0ed8332c702b0705/app/services/course_search_service.rb#L15

Reply:
fyi 

either approach it like
https://github.com/DFE-Digital/search-and-compare-api/blob/3aa6be82dcf5548892580e900ea1e37decf63a0f/src/api/DatabaseAccess/CourseDbContext.cs#L121-L127

in either case sub querying it will be easier

ie
```sql

select * from course where id in (select id from course where filter)
```

and move the ordering/distinct etc  to the outer query

Reply:
@TobyRet If the Stackoverflow Q you linked-to is correct, maybe you need to do:

```ruby
    scope = scope.distinct
```

before all the ordering.

Reply:
No unfortunately this did not work. An outer query had to be created instead.